### DISCUSSION TITLE
RGB-D Mapping in Outdoor Environments (SLAM or otherwise)

### DISCUSSION THREAD

**Discussion by kevinmsmith214** [Comet-Robotics/URC#14] [Autonomous]
Because the mapping will be outdoors, there are a few challenges that come along with it. Consequences of limited range of RGB-D sensors, lack of matched features in RGB frames, repetitive structures in environments, and the demand for rich detail are amplified in outdoor environments, and handling these issues has proved a challenging task.

 Here is a [paper](https://ieeexplore.ieee.org/document/8956916) that talks about implementing SLAM in an outdoor environment with the same realsense camera that we are using with some tweaks to overcome the challenges of being outside.

We need to research this further in the cases that SLAM is not used. Post any papers or links below.

**Comment by hectorv117** [comment]
Here's a good review [paper](https://springerplus.springeropen.com/articles/10.1186/s40064-016-3573-7) of different types of visual odometry and their pros/cons. This [paper](https://www-robotics.jpl.nasa.gov/media/documents/AAS_2020_mobility_mmc_v10.pdf) from jpl talks about how autonomy was implemented in the mars 2020 rover mission.

I think you explained the main challenges with mapping outdoor envs using rgbd really well. It seems that NASA primarily relies on stereo cameras for it's depth calculation and visual odometry. Which is good news for us since the d435 uses stereo cameras for depth calculation as well. I think the main way they use the stereo cameras to overcome a lot of the challenges that come with outdoor visual stereo based odometry is by primarily using the surface of mars as a way to minimize the distance away from the camera, and extract features and texture. So the success of rgbd mapping outdoors will be highly dependent on the surface of the outdoor environment we're using. It should preferably be a surface with a lot of detail/patterns maybe a gravel/dirt road with many scattered rocks even grass should work

**Comment by conner-replogle** [comment]
We plan to use the A* algorithm for pathfinding on the rover. A grid will be created for A*, with the starting position based on the current GPS and the target GPS as the end position. Initially, the grid will be empty, and the rover will follow a straight path. When an obstacle is encountered, a depth sensor will identify its location relative to the rover. The corresponding grid cell will then be marked as an obstacle, helping the A* algorithm to navigate around it. 

Challenges:
- Sizing obstacles using the depth data we need to create a boundary box of its size

First things first we need to simulate this in gazebo
